float64 x
float64 y
float64 velX
float64 velY
